### ISSUE TITLE
State functions

### ISSUE THREAD

**Issue by Y-Less** [TDW-org/samp-include-rscan#3]
I don't think (from looking at the code, I've not tested this) that this code won't work with state functions like:

```pawn
MyFunc() <my_state:a>
{
}

MyFunc() <my_state:b>
{
}

main()
{
    state my_state:b;
    MyFunc();
}
```

They are implemented as a call to a function *stub* near the start of the code block, which consists soley of a switch and jumps to individual complete functions.  The stub has no `PROC`, but any `CALL`s will point to it, not to the separate implementations.  I can provide better disassembly if you would like.

I like the idea though (does it detect its own recursion)?

**Comment by VVWVV** [comment]
Thank you!

I already knew about the problem but I don't know how to fix it. 

> (does it detect its own recursion)?

Yes.

**Comment by Y-Less** [comment]
Yeah, I'm not fully sure how to fix it, as even if you can detect the stub function (not hard), it then becomes like one `CALL` pointing to multiple functions and I don't know enough about your code to know how to deal with that.

Also, do you know about the tortoise and hare algorithm?  Used for detecting cycles in linked lists (which perfectly describes recursion):

https://en.wikipedia.org/wiki/Tortoise_and_hare_algorithm

May help, may not...

**Comment by VVWVV** [comment]
> Also, do you know about the tortoise and hare algorithm? Used for detecting cycles in linked lists (which perfectly describes recursion):
>
> https://en.wikipedia.org/wiki/Tortoise_and_hare_algorithm

I didn't even think about it. Thank you!
